backend/src/test/java/asegroup1/api/swagger/Swagger2MarkupTest.java

```diff
@@ -0,0 +1,57 @@
+package asegroup1.api.swagger;
+
+import asegroup1.api.Application;
+import asegroup1.api.configs.SwaggerConfig;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.io.BufferedWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebAppConfiguration
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = {Application.class, SwaggerConfig.class})
+@AutoConfigureMockMvc
+public class Swagger2MarkupTest {
+
```

Comment:
Yeah, I need to generate the swagger.json from the code, the simplest way to do it is as a part of the unit tests. Otherwise, I would have to add some compiled code to the repository to startup the application to get the swagger.json as a part of a post build phase